proto/v2/skaffold.proto

```diff
@@ -0,0 +1,819 @@
+    ActionableErr actionableErr = 5; // actionable error message
+}
+
+// `BuildSubtaskvent` describes the build status per artifact, and will be emitted by Skaffold anytime a build starts or finishes, successfully or not.
+// If the build fails, an error will be attached to the event.
+message BuildSubtaskEvent {
+    string id = 1; // id of the subtask which will be used in SkaffoldLog
+    string task_id = 2; // id of the task of skaffold that this event came from
+    string artifact = 3; // artifact name
+    string status = 4; // artifact build status oneof: InProgress, Completed, Failed
+    ActionableErr actionableErr = 5; // actionable error message
+}
+
+// `TestSubtaskEvent` represents the status of a test, and is emitted by Skaffold
+// anytime a test starts or completes, successfully or not.
+message TestSubtaskEvent {
+    string id = 1; // id of the subtask which will be used in SkaffoldLog
+    string task_id = 2; // id of the task of skaffold that this event came from
+    string status = 3; // test status oneof: InProgress, Completed, Failed
+    ActionableErr actionableErr = 4; // actionable error message
+}
+
+// `DeploySubtaskEvent` represents the status of a deployment, and is emitted by Skaffold
+// anytime a deployment starts or completes, successfully or not.
+message DeploySubtaskEvent {
+    string id = 1; // id of the subtask which will be used in SkaffoldLog
+    string task_id = 2; // id of the task of skaffold that this event came from
+    string status = 3; // deployment status oneof: InProgress, Completed, Failed
+    ActionableErr actionableErr = 4; // actionable error message
+}
+
+// A Resource StatusCheck Event, indicates progress for each kubernetes deployment.
+// For every resource, there will be exactly one event with `status` *Succeeded* or *Failed* event.
+// There can be multiple events with `status` *Pending*.
+// Skaffold polls for resource status every 0.5 second. If the resource status changes, an event with `status` “Pending”, “Complete” and “Failed”
+// will be sent with the new status.
+message StatusCheckSubtaskEvent {
+    string id = 1; // id of the subtask which will be used in SkaffoldLog
+    string task_id = 2; // id of the task of skaffold that this event came from
+    string resource = 3; // id of the subtask which will be used in SkaffoldLog
+    string status = 4; // id of the task of skaffold that this event came from
+    string message = 5;
+    StatusCode statusCode = 6;
+    ActionableErr actionableErr = 7; // actionable error message
+}
+
+// PortForwardEvent Event describes each port forwarding event.
+message PortForwardEvent {
+    string id = 1; // id of the subtask which will be used in SkaffoldLog
+    string task_id = 2; // id of the task of skaffold that this event came from
+    int32 localPort = 3; // local port for forwarded resource
+    int32 remotePort = 4; // Deprecated. Uses targetPort.intVal.
```

Comment:
remove deprecated event here and everywhere.

Reply:
Thanks for catching this, didn't realize I missed some